### PR TITLE
BLD: Fix nightlies

### PR DESCRIPTION
**Pull request by lithomas1**

### CODE REVIEW

File: azure/posix.yml [hunk 1]
```diff
@@ -63,6 +63,7 @@ jobs:
       - bash: |
           set -e
           pip install virtualenv wheel
+          pip install versioneer[toml]
           BUILD_DEPENDS="$NP_BUILD_DEP $CYTHON_BUILD_DEP"
```
Comment (datapythonista) [inline azure/posix.yml:67]: Maybe this:

```suggestion
          BUILD_DEPENDS="$NP_BUILD_DEP $CYTHON_BUILD_DEP versioneer[toml]"
```

Feels like the missing dependency is inside the docker image, and you're installing it in the CI image above.

**Review by datapythonista — COMMENTED** [review]
Sorry for the delay. Added a comment, but the CI logs aren't available anymore, so I'm guessing the failure in the CI is still the same, but maybe I'm wrong.

**Comment by datapythonista** [comment]
Seems like in the pandas repo we test with `numpy < 1.24`, and we're testing with numpy 1.24 here, I guess this is the cause of the CI errors?

**Review by datapythonista — APPROVED** [review]
Looks good. I'm not sure how the nightlies work, I guess the `BUILD_COMMIT` value is irrelevant, right? Because it'll be changed again once we release 1.5.3.

In any case, feel free to merge when you're happy with this.

**Comment by simonjayhawkins** [comment]
> I guess the `BUILD_COMMIT` value is irrelevant, right? Because it'll be changed again once we release 1.5.3.

IIUC the BUILD_COMMIT is used for the CI in the PR. So in the past I have tended to change to master to test for the nightly build then undo that change from the PR once green before merging.

**Comment by lithomas1** [comment]
Just for clarification, there are patches in this PR for both the nightlies (they broke a while back) and for 1.5.x.

I think the script in pandas-release
https://github.com/pandas-dev/pandas-release/blob/168489615c28b9326786edef6935a09582f26df0/scripts/wheels.sh#L25-L27

handles BUILD_COMMITs that don't start with a v correctly (e.g. branch), but not sure. I'll revert the BUILD_COMMIT changes and merge on red, though, just to be safe.

**Comment by simonjayhawkins** [comment]
> I'll revert the BUILD_COMMIT changes and merge on red, though, just to be safe.

sgtm

**Review by simonjayhawkins — APPROVED** [review]

**Pull request merged**